orchestra/todos/views.py

```diff
@@ -143,15 +143,23 @@ def delete(self, request, *args, **kwargs):
 
     @action(detail=False, methods=['put'])
     def put(self, request, *args, **kwargs):
+        partial = kwargs.get('partial', False)
         ids = [x['id'] for x in request.data]
-        instances = self.get_queryset(ids=ids)
+        # To avoid mess, sort data and queryset by ids
+        sorted_data = sorted(request.data, key=lambda x: x['id'])
+        instances = self.get_queryset(ids=ids).order_by('id')
         serializer = self.get_serializer(
-            instances, data=request.data, partial=False, many=True)
+            instances, data=sorted_data, partial=partial, many=True)
         serializer.is_valid(raise_exception=True)
         self.perform_update(serializer)
         data = serializer.data
         return Response(data)
 
+    @action(detail=False, methods=['patch'])
```

Comment:
do we need to do anything in a urls.py to expose this method?

Reply:
Nope :) `action` (and `MethodMapper`) takes care of urls mapping: https://github.com/encode/django-rest-framework/blob/f744da74d2878b480220ebaf9d8117ff9b79a947/rest_framework/decorators.py#L123